src/coreclr/jit/emit.h

```diff
@@ -2173,22 +2173,71 @@ class emitter
         return (emitCurIG && emitCurIGfreeNext > emitCurIGfreeBase);
     }
 
-    instrDesc* emitLastIns;
+#define EMIT_MAX_IG_INS_COUNT 256
+
+    // A recording of the last instructions, used for peephole optimizations.
+    // The max count of last instructions is the same as the max count of IG instructions.
+    instrDesc* emitLastInstrs[EMIT_MAX_IG_INS_COUNT];
+
+    inline instrDesc* emitGetLastIns() const
+    {
+        return emitLastInstrs[emitInsCount % ArrLen(emitLastInstrs)];
```

Comment:
```suggestion
        return emitLastInstrs[emitInsCount % EMIT_MAX_IG_INS_COUNT];
```

Reply:
I prefer `ArrLen` over the `#define` - it will still emit a constant either way. The reason being is that you only have to specify the length of `emitLastInstrs` once:
```cpp
instrDesc* emitLastInstrs[EMIT_MAX_LAST_INSTR_COUNT];
```